src/iwant_bot/pool.py

```diff
@@ -43,13 +45,20 @@ def update_requests_from_storage(self):
                 destination.append(req)
 
     def make_pairs(self, activity):
+        users = self.users
         source = self.req_by_activities[activity]
         destination = self.pairs[activity]
-        while len(source) > 1:
-            request1 = source.pop()
-            request2 = source.pop()
-            paired = self.pair(request1, request2)
-            destination.append(paired)
+        if len(source) > 1:
+            for request1 in source:
+                source.remove(request1)
```

Comment:
I will take a look at it. Soon. I'm thankful for your review. It was a bit unhappy and first working solution, but it's not perfect.